okhttp/src/main/java/com/squareup/okhttp/Connection.java

```diff
@@ -361,12 +277,17 @@ void setProtocol(Protocol protocol) {
     this.protocol = protocol;
   }
 
-  void setTimeouts(int readTimeoutMillis, int writeTimeoutMillis) throws IOException {
+  void setTimeouts(int readTimeoutMillis, int writeTimeoutMillis)
+      throws RouteException {
```

Comment:
I'm on board with RouteException. I was skeptical at first, but you've completely won me over. Good idea.